### PR TITLE
Edgcmnspr 30: Update to Java 17 (Update the module to Spring boot v3.0.1)

### PR DESCRIPTION
**Pull request by bekhzodkurbonboev**
## Purpose
[EDGCMNSPR-30](https://issues.folio.org/browse/EDGCMNSPR-30): edge-common-spring: Update to Java 17
[EDGCMNSPR-31](https://issues.folio.org/browse/EDGCMNSPR-31): edge-common-spring: Update the module to Spring boot v3.0.0 and identify issues.

## Approach
[Migrating to Spring Boot v3.0.0](https://wiki.folio.org/pages/viewpage.action?spaceKey=~Taras_Spashchenko&title=Migration+to+Spring+Boot+v3.0.0+How+to)

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [ ] Duplications on new code is 3% or less
   - [ ] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by TarasSpashchenko** [comment]
<version>2.1.1-SNAPSHOT</version>
Please bump up the artifact version to the next major release

**Review by TarasSpashchenko — CHANGES_REQUESTED** [review]
2.1.1-SNAPSHOT
Please bump up the artifact version to the next major release

File: src/main/java/org/folio/edgecommonspring/filter/EdgeSecurityFilter.java [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/edgecommonspring/filter/EdgeSecurityFilter.java:12]: Looks like this module doesn't have checkstlyle, so build doesn't fail. Nevertheless, could you please use alphabetic order for imports to be consistent with existing code?

Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/edgecommonspring/filter/EdgeSecurityFilter.java:10]: Thank you! I have refactored the order of imports.

**Review by TarasSpashchenko — APPROVED** [review]

**Review by alekGbuz — APPROVED** [review]

**Review by khandramai — APPROVED** [review]

**Pull request merged**